frontend/src/lib/k8s/api/v2/webSocket.ts

```diff
@@ -167,3 +167,96 @@ export function useWebSocket<T>({
     };
   }, [enabled, url, protocols, type, cluster]);
 }
+
+export type WebSocketConnectionRequest<T> = {
+  cluster: string;
+  url: string;
+  onMessage: (data: T) => void;
+};
+
+/**
+ * Creates or joins mutiple existing WebSocket connections
+ *
+ * @param url - endpoint URL
+ * @param options - WebSocket options
+ */
+export function useWebSockets<T>({
+  connections,
+  enabled = true,
+  protocols,
+  type = 'json',
+}: {
+  enabled?: boolean;
+  connections: Array<WebSocketConnectionRequest<T>>;
+  /**
+   * Any additional protocols to include in WebSocket connection
+   */
+  protocols?: string | string[];
+  /**
+   *
+   */
+  type?: 'json' | 'binary';
+  /**
+   * Message callback
+   */
+}) {
+  useEffect(() => {
+    if (!enabled) return;
```

Comment:
Nit: We don't have a lot of in line conditionals in Headlamp. Apparently we don't enforce/lint it but we do the full body conditionals normally.

Reply:
adjusted